### PR TITLE
Orchestrator, Dispatcher, Sequencer integration

### PR DESCRIPTION
**Pull request by samparent97**
Integrates orchestrator, sequencer and the cli dispatcher. Also confirmed that multiple dispatchers work as long as the dispatcher monitors the orchestrator signals (status and results). Check monitorOrchestrator() on any dispatcher implementation to see an example.

Adds:
- Orchestrator
- Dispatcher
- A bunch of test States, Sequences, Fake dispatchers, result accumulators

Edits:
- Sequencer

### CODE REVIEW

**Review by itshady — APPROVED** [review]
Still need another approval since I worked on this

File: orchestrator/orchestrator.go [hunk 1]
```diff
@@ -2,67 +2,329 @@ package orchestrator
...
+		progCh:            make(chan flow.Progress),
+		cancelCurrentTest: make(chan struct{}),
+		testQueueMtx:      sync.Mutex{},
+		progressMtx:       sync.Mutex{},
+		fatalErr:          utils.NewResettaleError(),
 	}
 
-	for _, o := range opts {
-		o(ret)
+	for _, d := range dispatchers {
```
Comment (samparent97) [inline orchestrator/orchestrator.go:68]: ret.dispatchers = dispatchers

File: orchestrator/orchestrator.go [hunk 2]
```diff
@@ -2,67 +2,329 @@ package orchestrator
...
+		go o.monitorDispatcher(ctx, d)
+	}
 
 	return nil
 }
 
-func (o *Orchestrator) Start(ctx context.Context) error {
-	return nil
+func (o *Orchestrator) Run(ctx context.Context) error {
+	const _checkForStartSignalPeriod = 100 * time.Millisecond
```
Comment (samparent97) [inline orchestrator/orchestrator.go:112]: make this 50 ms

File: orchestrator/orchestrator.go [hunk 3]
```diff
@@ -2,67 +2,329 @@ package orchestrator
...
+		}
+
+		if o.state == FatalError {
+			continue
+		}
+
+		// Attempt to dequeue next test, if no test queued continue.
+		startSig, ok := o.dequeueNextTest()
+		if !ok {
+			if o.state == Running {
```
Comment (samparent97) [inline orchestrator/orchestrator.go:133]: add comment to explain this logic

**Review by hsidd1 — APPROVED** [review]
magnificient

**Review by TylerStAmour — APPROVED** [review]
LGTM

**Pull request merged**